File: src/main/java/com/android/helper/utils/LogUtil.java
package com.android.helper.utils;

import android.text.TextUtils;

import com.android.helper.app.BaseApplication;
import com.android.helper.common.CommonConstants;
import com.orhanobut.logger.Logger;

/**
 * @author XJX  日志工具类
 */
public class LogUtil {

    private static LogWriteUtil writeUtil;

    public static void e(String value) {
        if (!TextUtils.isEmpty(value)) {
            if (isDebug()) {
                Logger.e(value);
            }
        }
    }

    public static void e(String tag, String value) {
        if (!TextUtils.isEmpty(value)) {
            if (isDebug()) {
                Logger.t(tag).e(value);
            }
        }
    }

    public static void d(String value) {
        if (!TextUtils.isEmpty(value)) {
            if (isDebug()) {
                Logger.d(value);
            }
        }
    }

    public static void d(String tag, String value) {
        if (!TextUtils.isEmpty(value)) {
            if (isDebug()) {
                Logger.t(tag).d(value);
            }
        }
    }

    public static void i(String value) {
        if (!TextUtils.isEmpty(value)) {
            if (isDebug()) {
                Logger.i(value);
            }
        }
    }

    public static void i(String tag, String value) {
        if (!TextUtils.isEmpty(value)) {
            if (isDebug()) {
                Logger.t(tag).i(value);
            }
        }
    }

    public static void w(String value) {
        if (!TextUtils.isEmpty(value)) {
            if (isDebug()) {
                Logger.w(value);
            }
        }
    }

    public static void w(String tag, String value) {
        if (!TextUtils.isEmpty(value)) {
            if (isDebug()) {
                Logger.t(tag).w(value);
            }
        }
    }

    /**
     * 专门用来监听邦车逻辑的log
     *
     * @param msg 具体的消息
     */
    public static void car(String msg) {
        if (isDebug()) {
            Logger.t("BindCar").e(msg);
        }
    }

    /**
     * 在debug模式下会把日志写入到sd卡中
     *
     * @param fileName 文件的名字,例如：电车数据 ，这里不用带格式名字，方法会自动拼写。
     * @param value    写入的内容
     */
    public static void writeDe(String fileName, String value) {
        if ((!TextUtils.isEmpty(fileName) && (!TextUtils.isEmpty(value)))) {
            if (isDebug()) {
                if (writeUtil == null) {
                    writeUtil = new LogWriteUtil();
                }
                writeUtil.write(fileName, value);
            }
        }
    }

    /**
     * 写入保活的信息日志
     */
    public static void writeLifeCycle(String value) {
        if (!TextUtils.isEmpty(value)) {
            if (writeUtil == null) {
                writeUtil = new LogWriteUtil();
            }
            writeUtil.write(CommonConstants.FILE_LIFECYCLE_NAME, value);
            if (isDebug()) {
                Logger.e("应用保活：" + value);
            }
        }
    }

    /**
     * 设置充电中心的数据
     *
     * @param value 具体的数据
     */
    public static void writeChargingCenter(String value) {
        if (!TextUtils.isEmpty(value)) {
            if (writeUtil == null) {
                writeUtil = new LogWriteUtil();
            }
            writeUtil.write(CommonConstants.FILE_CHARGING_CENTER_NAME, value);
            if (isDebug()) {
                Logger.e("充电中心：" + value);
            }
        }
    }

    /**
     * 写入保活的信息日志
     */
    public static void write(String fileName, String value) {
        if ((!TextUtils.isEmpty(value)) && (!TextUtils.isEmpty(fileName))) {
            if (writeUtil == null) {
                writeUtil = new LogWriteUtil();
            }
            writeUtil.write(fileName, value);
        }
    }

    /**
     * @return 检测当前时候是debug模式，默认不是
     */
    public static boolean isDebug() {
        boolean debug = false;
        BaseApplication application = BaseApplication.getInstance();
        if (application != null) {
            debug = application.isDebug();
        }
        return debug;
    }

}

File: src/main/java/com/android/helper/utils/LogUtil.kt
package com.android.helper.utils

import android.text.TextUtils
import com.android.helper.app.BaseApplication
import com.orhanobut.logger.Logger

/**
 * @author XJX 日志工具类
 */
object LogUtil {

    private val mWriteUtil: LogWriteUtil = LogWriteUtil()

    @JvmStatic
    fun e(tag: String = "", value: Any?) {
        if (isDebug) {
            value?.let {
                val contentValue = if (value is String) {
                    value
                } else {
                    value.toString()
                }

                if (!TextUtils.isEmpty(contentValue)) {
                    if (TextUtils.isEmpty(tag)) {
                        Logger.e(contentValue)
                    } else {
                        Logger
                            .t(tag)
                            .e(contentValue)
                    }
                }
            }
        }
    }

    @JvmStatic
    fun e(value: Any?) {
        e(value = value, tag = "")
    }

    @JvmStatic
    fun d(tag: String = "", value: Any?) {
        if (isDebug) {
            value?.let {
                val contentValue = if (value is String) {
                    value
                } else {
                    value.toString()
                }

                if (!TextUtils.isEmpty(contentValue)) {
                    if (TextUtils.isEmpty(tag)) {
                        Logger.d(contentValue)
                    } else {
                        Logger
                            .t(tag)
                            .d(contentValue)
                    }
                }
            }
        }
    }

    @JvmStatic
    fun d(value: Any?) {
        d(value = value, tag = "")
    }

    @JvmStatic
    fun i(tag: String = "", value: Any?) {
        if (isDebug) {
            value?.let {
                val contentValue = if (value is String) {
                    value
                } else {
                    value.toString()
                }

                if (!TextUtils.isEmpty(contentValue)) {
                    if (TextUtils.isEmpty(tag)) {
                        Logger.i(contentValue)
                    } else {
                        Logger
                            .t(tag)
                            .i(contentValue)
                    }
                }
            }
        }
    }

    fun i(value: Any?) {
        i(value = value, tag = "")
    }

    @JvmStatic
    fun w(tag: String = "", value: Any?) {
        if (isDebug) {
            value?.let {
                val contentValue = if (value is String) {
                    value
                } else {
                    value.toString()
                }

                if (!TextUtils.isEmpty(contentValue)) {
                    if (TextUtils.isEmpty(tag)) {
                        Logger.w(contentValue)
                    } else {
                        Logger
                            .t(tag)
                            .w(contentValue)
                    }
                }
            }
        }
    }

    @JvmStatic
    fun w(value: Any?) {
        w(value = value, tag = "")
    }

    /**
     * write file content
     */
    @JvmStatic
    @JvmOverloads
    fun write(fileName: String? = "", value: String? = "") {
        if ((!TextUtils.isEmpty(value)) && (!TextUtils.isEmpty(fileName))) {
            mWriteUtil.write(fileName!!, value)
        }
    }

    /**
     * write  file content and write log
     */
    @JvmStatic
    fun writeAll(fileName: String? = "", tag: String? = "", value: String? = "") {
        if ((!TextUtils.isEmpty(value)) && (!TextUtils.isEmpty(fileName))) {
            value?.let {
                if (TextUtils.isEmpty(tag)) {
                    Logger.e(it)
                } else {
                    Logger
                        .t(tag)
                        .e(it)
                }
                mWriteUtil.write(fileName!!, it)
            }
        }
    }

    @JvmStatic
    fun writeAll(fileName: String? = "", value: String? = "") {
        writeAll(fileName = fileName, value = value, tag = "")
    }

    /**
     * @return if application type is debug,return true,else return false
     */
    private val isDebug: Boolean
        get() {
            val application = BaseApplication.getInstance()
            application?.let {
                return it.isDebug
            }
            return false
        }

}

